test/transpile.jl

```diff
@@ -760,13 +840,61 @@ end
     end
 end
 
-@testset "cast_to_cz: iswap" begin
+@testset "CastCXToCZGateTranspiler: Readout" begin
+    transpiler = CastCXToCZGateTranspiler()
+
+    circuits = [
+        QuantumCircuit(qubit_count = 2, instructions = [Readout(1)]),
+        QuantumCircuit(qubit_count = 2, instructions = [control_x(1, 2), Readout(1)]),
+        QuantumCircuit(
+            qubit_count = 2,
+            instructions = [control_x(1, 2), x_90(1), control_x(1, 2), Readout(1)],
+        ),
+        QuantumCircuit(
+            qubit_count = 2,
+            instructions = [iswap(1, 2), control_x(1, 2), Readout(1)],
+        ),
+    ]
+
+    for circuit in circuits
+        transpiled_circuit = transpile(transpiler, circuit)
+
+        @test !circuit_contains_gate_type(transpiled_circuit, Snowflurry.ControlX)
+        @test compare_circuits(circuit, transpiled_circuit)
+    end
+end
+
+@testset "CastISwapToCZGateTranspiler: iswap" begin
+    transpiler = CastISwapToCZGateTranspiler()
+
+    circuits = [
+        QuantumCircuit(qubit_count = 2, instructions = [iswap(1, 2)]),
+        QuantumCircuit(qubit_count = 2, instructions = [iswap(1, 2), x_90(1), iswap(1, 2)]),
+        QuantumCircuit(qubit_count = 2, instructions = [control_x(1, 2), iswap(1, 2)]),
+    ]
+
+    for circuit in circuits
+        transpiled_circuit = transpile(transpiler, circuit)
+
+        @test !circuit_contains_gate_type(transpiled_circuit, Snowflurry.ISwap)
+        @test compare_circuits(circuit, transpiled_circuit)
```

Comment:
Same as above

Reply:
updated `compare_circuits` to check `Readouts`